src/colorizer/TimeControls.ts

```diff
@@ -0,0 +1,132 @@
+import ColorizeCanvas from "./ColorizeCanvas";
+
+// time / playback controls
+export default class TimeControls {
+  private playBtn: HTMLButtonElement;
+  private pauseBtn: HTMLButtonElement;
+  private forwardBtn: HTMLButtonElement;
+  private backBtn: HTMLButtonElement;
+  private timeSlider: HTMLInputElement;
+  private timeInput: HTMLInputElement;
+
+  private timerId: number;
+  private redrawfn: () => void;
+
+  private canvas: ColorizeCanvas;
+  private isDisabled: boolean;
+
+  constructor(canvas: ColorizeCanvas, redrawfn: () => void) {
+    this.redrawfn = redrawfn;
+    this.canvas = canvas;
+    this.timerId = 0;
+    this.isDisabled = false;
+
+    this.playBtn = document.querySelector("#playBtn")!;
+    this.pauseBtn = document.querySelector("#pauseBtn")!;
+    this.forwardBtn = document.querySelector("#forwardBtn")!;
+    this.backBtn = document.querySelector("#backBtn")!;
+    this.timeSlider = document.querySelector("#timeSlider")!;
+    this.timeInput = document.querySelector("#timeValue")!;
+    this.playBtn.addEventListener("click", () => this.handlePlayButtonClick());
+    this.pauseBtn.addEventListener("click", () => this.handlePauseButtonClick());
+    this.forwardBtn.addEventListener("click", () => this.handleFrameAdvance(1));
+    this.backBtn.addEventListener("click", () => this.handleFrameAdvance(-1));
+    // only update when DONE sliding: change event
+    this.timeSlider.addEventListener("change", () => this.handleTimeSliderChange());
+    this.timeInput.addEventListener("change", () => this.handleTimeInputChange());
```

Comment:
Worth thinking about generalizing this class and `RecordingControls` to just hold handlers and letting main.ts handle the arbitrary business of getting DOM controls and binding the handlers to them. Out of scope for this PR though.

Reply:
I like that idea, similar in spirit to #30 with the goal of decoupling UI from business logic.